tests/simulator_test.cpp

```diff
@@ -2,82 +2,206 @@
+
+  Ray atSixtyOther(kVecZero,
+                   Vec3(-std::cos(deg2rad(60)), 0, std::sin(deg2rad(60))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, atSixtyOther, &hitData));
+
+  Ray atSixtyXY(kVecZero, Vec3(std::cos(deg2rad(60)), std::cos(deg2rad(60)),
+                               std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, atSixtyXY, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray atSixtyXYOther(kVecZero,
+                     Vec3(-std::cos(deg2rad(60)), -std::cos(deg2rad(60)),
+                          std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, atSixtyXYOther, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+}
+
+TEST_F(EnergyCollectorTest, EvenNumOfEnergyCollectorTest) {
+  const FakeModel nonEmptyModel(false);
+
+  const int numCollectors = 36;
+  auto energyCollectors = buildCollectors(nonEmptyModel, numCollectors);
+  ASSERT_EQ(energyCollectors.size(), numCollectors);
+
+  Ray straightUp(kVecZero, kVecUp);
+  RayHitData hitData;
+  ASSERT_TRUE(performHitCollector(energyCollectors, straightUp, &hitData));
+
+  Ray straightDown(kVecZero, -kVecUp);
+  ASSERT_FALSE(performHitCollector(energyCollectors, straightDown, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray alongX(kVecZero, kVecX);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
+
+  const float collectorPositionRadius = 4;
+  const float collectorAngle = 2 * kPi / (numCollectors - 2);
+  const float refCollectorRadius =
+      collectorPositionRadius * std::sqrt(2 - 2 * std::cos(collectorAngle));
+  ASSERT_EQ(alongX.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray alongY(kVecZero, kVecY);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
+
+  ASSERT_EQ(alongY.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray at30(kVecZero, Vec3(std::cos(deg2rad(30)), 0, std::sin(deg2rad(30))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, at30, &hitData));
+
+  Ray at30other(kVecZero,
+                Vec3(-std::cos(deg2rad(30)), 0, std::sin(deg2rad(30))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, at30other, &hitData));
+
+  Ray at30XY(kVecZero, Vec3(std::cos(deg2rad(30)), std::cos(deg2rad(30)),
+                            std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, at30XY, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray at30XYOther(kVecZero, Vec3(-std::cos(deg2rad(30)), -std::cos(deg2rad(30)),
+                                 std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, at30XYOther, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+}
+
+TEST_F(EnergyCollectorTest, PositionsThatWereFixedTest) {
```

Comment:
Name of the test does not explain what are those cases. Make sure to name tests so that they describe what the test is about.

Reply:
is this name better?